mods/hide-dotfiles-explorer.wh.cpp

```diff
@@ -0,0 +1,290 @@
+                                    shouldInclude = true;
+                                }
+                            }
+                        }
+                        CoTaskMemFree(pidlAbs);
+                    }
+                    CoTaskMemFree(pidlFolder);
+                }
+                pPersist->Release();
+            }
+            if (pszFullPath) CoTaskMemFree(pszFullPath);
+            if (shouldInclude) {
+                rgelt[fetched++] = pidl;
+            } else {
+                CoTaskMemFree(pidl);
+            }
+        }
+        if (pceltFetched) {
+            *pceltFetched = fetched;
+        }
+        return (fetched == celt) ? S_OK : S_FALSE;
+    }
+    STDMETHODIMP Skip(ULONG celt) {
+        if (!m_pOriginal) return E_FAIL;
+        return m_pOriginal->Skip(celt);
+    }
+    STDMETHODIMP Reset() {
+        if (!m_pOriginal) return E_FAIL;
+        return m_pOriginal->Reset();
+    }
+    STDMETHODIMP Clone(IEnumIDList** ppenum) {
+        if (!ppenum) return E_INVALIDARG;
+        *ppenum = nullptr;
+        if (!m_pOriginal) return E_FAIL;
+        IEnumIDList* pClonedOriginal = nullptr;
+        HRESULT hr = m_pOriginal->Clone(&pClonedOriginal);
+        if (SUCCEEDED(hr) && pClonedOriginal) {
+            DotfileFilterEnum* pClonedWrapper = new (std::nothrow) DotfileFilterEnum(pClonedOriginal, m_pFolder);
+            pClonedOriginal->Release();
+            if (pClonedWrapper) {
+                *ppenum = pClonedWrapper;
+                return S_OK;
+            }
+            return E_OUTOFMEMORY;
+        }
+        return hr;
+    }
+};
+
+HRESULT WINAPI IShellFolder_EnumObjects_Hook(IShellFolder* pThis, HWND hwnd, SHCONTF grfFlags, IEnumIDList** ppenumIDList) {
+    HRESULT hr = IShellFolder_EnumObjects_Original(pThis, hwnd, grfFlags, ppenumIDList);
+    if (SUCCEEDED(hr) && ppenumIDList && *ppenumIDList) {
+        IEnumIDList* pOriginal = *ppenumIDList;
+        DotfileFilterEnum* pFiltered = new (std::nothrow) DotfileFilterEnum(pOriginal, pThis);
+        if (pFiltered) {
+            *ppenumIDList = pFiltered;
+            pOriginal->Release();
+        } else {
+            pOriginal->Release();
+            *ppenumIDList = nullptr;
+            hr = E_OUTOFMEMORY;
+        }
+    }
+    return hr;
+}
+
+void LoadSettings() {
+    PCWSTR excludePaths = Wh_GetStringSetting(L"excludePaths");
+    ParseExcludePaths(excludePaths);
+    if (excludePaths) {
+        Wh_FreeStringSetting(const_cast<WCHAR*>(excludePaths));
+    }
+}
+
+BOOL Wh_ModInit() {
+    LoadSettings();
+    
+    HMODULE hShell32 = GetModuleHandle(L"shell32.dll");
+    if (!hShell32) return FALSE;
+    
+    void* pEnumObjects = nullptr;
+    for (int ordinal = 1; ordinal < 100; ordinal++) {
+        FARPROC pFunc = GetProcAddress(hShell32, (LPCSTR)ordinal);
+        if (pFunc) {
+            BYTE* pBytes = (BYTE*)pFunc;
+            if (pBytes[0] == 0x48 && pBytes[1] == 0x89 && pBytes[2] == 0x5C) {
```

Comment:
This doesn't look very reliable, why was it added? Why not always use `SHGetDesktopFolder` like it was done previously?